pom.xml

```diff
@@ -24,7 +24,7 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <java.version>1.6</java.version>
+    <java.version>1.7</java.version>
```

Comment:
Thank you for reviewing.

About the formatting, yes I'm sorry, my bad, this might be my IDE's fault, and yes, I would apply the changes on naming that violets naming rules(sorry didn't know there is one)

The main goal is about recycling/reusing an instance `Json[Reader|Writer]`, by turn it from an immutable, to mutable one. This is actually could be marked as performance-benefit, by recycling an instance, instead of mark it for wipe(GC), and create another one(but not essentially all the time, e.g. small apps).

The basic `JsonReader`, and `JsonWriter` are still in immutable aspect, but the changes have applied now allow the mutability for sub-types (`MutableJsonWriter`, and `MutableJsonReader`).

Both `MutableJsonWriter`, and `MutableJsonReader` featuring:  

* Mutability, now they can be reset, to be used for another read/write op, without creating a new instance.
* Allowing to skip closing the underlying io reader/writer by closing the json instance.
* Allowing to set(without state reset) and get underlying reader/writer during json read/write op(may help in a very uncommon situations)

Reply:
> The main goal is about recycling/reusing an instance `Json[Reader|Writer]`, by turn it from an immutable, to mutable one. This is actually could be marked as performance-benefit, by recycling an instance, instead of mark it for wipe(GC), and create another one(but not essentially all the time, e.g. small apps).

That sounds like premature optimization. Creating `JsonReader` objects is probably fairly cheap because they do not store that much data, and `JsonWriter` objects store nearly no data. Have you noticed performance issues when using these classes?

Reply:
Yes, you are right.

Actually not a very critical performance hit for sure, as the `Json[Writer|Reader]` is light in heart.

Depending on using a mutable type, this could be a con(as performance hit) actually, for instance acquiring the obj from a `synchronized` context.

As you mentioned, for sure, there is no any big win to have a mutable type of reader/writer, except if object recycling is an essential policy.